annotations/update_pending_annotations.py

```diff
@@ -0,0 +1,152 @@
+import synapseclient
+from synapseclient import Table, RowSet
+import argparse
+import pandas as pd
+from attribute_dictionary import PUBLICATION_DICT
+
+
+### Login to Synapse ###
+def login():
+
+    syn = synapseclient.Synapse()
+    syn.login()
+
+    return syn
+
+
+def get_args():
+
+    parser = argparse.ArgumentParser(
+        description=
+        'Get synapse table id of annotations to be editd and synapse table id of controlled vocabulary mappings'
+    )
+    parser.add_argument(
+        '-t',
+        '--table_id',
+        type=str,
+        default='syn21868591',
+        help=
+        'Synapse table id where annotations will be updated. Probably the Publications Merged table.'
+    )
+    parser.add_argument('-m',
+                        '--manifest_path',
+                        type=str,
+                        help='Path where updated manifest is stored.')
+    return parser.parse_args()
+
+
+def get_updated_df(manifest):
+
+    updated_df = pd.read_csv(manifest, index_col=False).fillna("")
+
+    return updated_df
+
+
+def get_annotations(table_id, updated_df, syn):
+
+    pubmed_list = updated_df['Pubmed Id'].astype(str).tolist()
+    pubmed_string = ', '.join(pubmed_list)
+    print(f'list of pumbmeds to be updated: {pubmed_string}')
+
+    annots_query = syn.tableQuery(
+        f"SELECT pubMedId, assay, tumorType, tissue, dataset, accessibility FROM {table_id} WHERE pubMedId IN ({pubmed_string})"
+    )
+
+    return annots_query
+
+
+def edit_annotations(updated_df, annots_query, syn, table_id):
+
+    # Convert annotations to data frame
+    annots_df = annots_query.asDataFrame().fillna("")
+    # preserve original index
+    index_rows = dict(zip(annots_df.pubMedId, annots_df.index))
+
+    # rename updated_df columns to match annots_df columns
+    updated_df.rename(columns=PUBLICATION_DICT, inplace=True)
+    updated_df.drop(['Component'], axis=1, inplace=True)
+
+    # update annots_df to match updated_df
+    final_df = annots_df.set_index('pubMedId')
+    updated_df = updated_df.set_index('pubMedId')
+    final_df.update(updated_df)
+    # resetting index so pubMedId can be used to map original index
+    final_df.reset_index(inplace=True)
+    # Reset index to match original annots_df using index_rows dict.
+    final_df['index'] = final_df['pubMedId'].map(index_rows)
+    final_df.set_index(keys='index', inplace=True)
+
+    # Get column data types from synapse table
+    cols = syn.getTableColumns(table_id)
+    col_dict = {}
+    for col in cols:
+        for k, v in col.items():
```

Comment:
Since you're only interested in the `name` key, you can forgo the dictionary iteration altogether and simplify this to:

```
for col in cols:
   col_name = col['name']
   col_dict[col_name] = col['columnType']
```

You don't have to worry about optimizations for this script, but keep in mind that loops take longer and longer to run through, the more items there are.

Reply:
Ah gotcha!